Update README.md (#279)
huggingface/safetensors: README.md

- 8 bytes: `N`, a u64 int, containing the size of the header
- N bytes: a JSON utf-8 string representing the header.
  - The header is a dict like `{"TENSOR_NAME": {"dtype": "F16", "shape": [1, 16, 256], "offsets": [BEGIN, END]}, "NEXT_TENSOR_NAME": {...}, ...}`, where offsets point to the tensor data relative to the beginning of the byte buffer, with `BEGIN` as the starting offset and `END` as the one-past offset (so total tensor byte size = `END - BEGIN`).

>   - A special key `__metadata__` is allowed to contain free form text-to-text map. Currently, only string values are allowed in `__metadata__`, arbitrary JSON is not allowed.

Reviewer: ```suggestion
  - A special key `__metadata__` is allowed to contain free form string-to-string map. Arbitrary JSON is not allowed, all values must be strings.
```
maybe like this is clearer

Reviewer 2: I prefer that phrasing very slightly too. Not strong opinion.

Tests will fail (because that document is replicated in a few places and there are checks to make sure they are in sync). I'll update when we agree on the phrasing.
